README.md

```diff
@@ -1,19 +1,45 @@
 # XML Component [![CircleCI](https://circleci.com/gh/elasticio/xml-component.svg?style=svg)](https://circleci.com/gh/elasticio/xml-component)
 
 ## Description
-elastic.io iPaaS component to convert between XML and JSON data. 
+iPaaS component to convert between XML and JSON data. 
 
 ### Purpose
-Allows users to convert XML attachments and strings to and from JSON
-This component has 3 actions allowing users to pass in either generic but well format XML/JSON string or XML attachment 
-and produces a generic string of the other file type. The output then can be maped and used in other components.
+Allows users to convert XML attachments and strings to and from JSON.
+This component has 3 actions allowing users to pass in either generic but well formatted XML/JSON strings or XML attachments 
+and produces a generic string or attachment of the other file type. The output then can be mapped and used in other components.
 
-### Requirements
+### Requirements and Conversion Behavior
 Provided XML document (for `XML to JSON`) should be [well-formed](https://en.wikipedia.org/wiki/Well-formed_document) 
-in order to be parsed correctly. You will get an error otherwise.
+in order to be parsed correctly. You will get an error otherwise. 
+
+JSON inputs must be objects with exactly one field as XML documents must be contained in a single 'root' tag. 
+[JSON inputs can not have any field names which are not valid as XML tag names:](https://www.w3schools.com/xml/xml_elements.asp) 
+* They must start with a letter or underscore
+* They cannot start with the letters xml (or XML, or Xml, etc)
+* They must only contain letters, digits, hyphens, underscores, and periods
+
+XML attributes on a tag can be read and set by setting an `_attr` sub-object in the JSON.  
+The inner-text of an XML element can also be controlled with `#` sub-object.
+
+For example:
+```json
+{
+  "someTag": {
+    "_attr": {
+      "id": "my id"
+    },
+    "_": "my inner text"
```

Comment:
So '_' or '#'?

Reply:
[As per the documentation `charKey` defaults to `_` though it was `#` in earlier versions.](https://www.npmjs.com/package/xml2js#options) We can change it to whatever we want though...